hooks/useIsMobile.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from 'react'
+
+export const useIsMobile = () => {
+    const [isMobile, setIsMobile] = useState<boolean>(false)
+
+    useEffect(() => {
+        const handleIsMobile = () => {
+            setIsMobile(window.innerWidth <= 1024)
+        }
+        handleIsMobile()
+        if (process.env.NODE_ENV === 'development') {
+            window.addEventListener('resize', handleIsMobile)
+            return () => window.removeEventListener('resize', handleIsMobile)
```

Comment:
[Enhancement]
Put `return` function inside condition is OK. But you should put this `return` outside the condition for avoiding confuse if your code has multiple condition check later.

Example:

```
if (process.env.NODE_ENV === 'development') {
           ...
}
 return () => {
    if (condition here) ...
 }
 ```

reference that post:
https://stackoverflow.com/questions/59890480/can-different-cleanups-be-done-conditionally-in-useeffect#:~:text=Add%20a%20comment-,2%20Answers,-Active